### PR TITLE
First do it

### PR DESCRIPTION
**Pull request by alexfedoseev**
Now it renders the view, but we should discuss a lot of stuff (:

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,3 +23,6 @@ npm-debug.log
 
 # Ignore bundle dependencies
 vendor/ruby
+
+# RVM gemset
+.ruby-gemset
```
Comment (justin808) [inline .gitignore]: Curious if this should be project specific.

File: client/assets/javascripts/actions/CommentActionCreators.js [hunk 2]
```diff
@@ -1,61 +1,61 @@
...
+  };
 }
 
 export function submitComment(comment) {
-  return function(dispatch) {
+  return dispatch => {
     dispatch(incrementAjaxCounter());
     return CommentsManager.submitComment(comment).then(
-        comment => dispatch(submitCommentSuccess(comment),
+        commentFromServer => dispatch(submitCommentSuccess(commentFromServer)),
```
Comment (justin808) [inline client/assets/javascripts/actions/CommentActionCreators.js]: :+1:

File: client/assets/javascripts/actions/CommentActionCreators.js [hunk 3]
```diff
@@ -1,61 +1,61 @@
...
       comments => dispatch(fetchCommentsSuccess(comments)),
       error => dispatch(fetchCommentsFailure(error))
     );
-  }
+  };
 }
 
 export function submitComment(comment) {
-  return function(dispatch) {
+  return dispatch => {
```
Comment (justin808) [inline client/assets/javascripts/actions/CommentActionCreators.js]: :+1:

File: client/assets/javascripts/components/CommentList.jsx [hunk 4]
```diff
@@ -10,12 +10,12 @@ const CommentList = React.createClass({
 
   render() {
     const reversedData = this.props.comments.slice(0).reverse();
-    const commentNodes = reversedData.map((comment, index) => {
+    const commentNodes = this.props.comments.get('comments').map((comment, index) => {
```
Comment (justin808) [inline client/assets/javascripts/components/CommentList.jsx]: Need to sort out the reversed data. The reducer is putting the new one at that the top. I'm not sure if we should have the store in reverse order, and not reverse for display or vice versa.

File: client/webpack.common.config.js [hunk 5]
```diff
@@ -13,8 +13,10 @@ module.exports = {
   //   jquery: 'var jQuery'
   // },
   resolve: {
-    root: [path.join(__dirname, 'scripts'),
-           path.join(__dirname, 'assets/javascripts'),],
+    root: [
```
Comment (justin808) [inline client/webpack.common.config.js]: In general, best to put white space changes in separate PR.

File: client/assets/javascripts/stores/CommentStore.js [hunk 6]
```diff
@@ -1,7 +1,10 @@
-import { createStore, applyMiddleware } from 'redux';
-import commentReducer from '../reducers/commentReducer';
+import { createStore, applyMiddleware, combineReducers } from 'redux';
+import thunk from 'redux-thunk';
+import reducers from '../reducers';
+
+// Smth wrong with this one, off for now
 import loggerMiddleware from '../middleware/loggerMiddleware';
```
Comment (justin808) [inline client/assets/javascripts/stores/CommentStore.js]: Yes -- let's see if we can get this one working afterwards, maybe a separate PR.

**Comment by justin808** [comment]
@alexfedoseev Looks AWESOME. I'll let you merge this change into the PR on my repo. Please respond to my comments here, however. Please merge --squash your changes on top of mine.

File: .gitignore (hunk 1 above)
Comment (alexfedoseev) [inline .gitignore]: I keep all gems in isolated gemsets for two reasons:

1. I maintain `yo` generator, so I scaffold a lot for testing purposes. It's easier for me to delete scaffolded test gems by deleting gemset.
1. To sleep easily.

File: client/webpack.common.config.js (hunk 5 above)
Comment (alexfedoseev) [inline client/webpack.common.config.js]: Agreed.

File: client/assets/javascripts/stores/CommentStore.js (hunk 6 above)
Comment (alexfedoseev) [inline client/assets/javascripts/stores/CommentStore.js]: Will take a look into this one.

Comment (alexfedoseev) [inline client/assets/javascripts/stores/CommentStore.js]: Fixed.

File: .gitignore (hunk 1 above)
Comment (justin808) [inline .gitignore]: Should .ruby-gemset be in developer's global gitignore?

Comment (alexfedoseev) [inline .gitignore]: > Should .ruby-gemset be in developer's global gitignore?

If **everybody** is using gemset-per-app strategy, then no. Otherwise — yes.

**Comment by justin808** [comment]
Already merged to other PR. Closing.
